### ISSUE TITLE
[1.16.5] Minecraft crashing randomly on Multiplayer server.

### ISSUE THREAD

**Issue by HauntedSniper** [sp614x/optifine#5379]
## Description of Issue
Seemingly at random, the server will just start reloading (as if reloading textures) and then crash me. I also crashed one time when trying to change perspective with F5, and most of the times that I try to change or move resource packs. The server in question is provided by Aternos, with two players being online. The server has a resourcepack that includes cit textures.

Rarely, but it does happen, will the game crash even when I'm playing alone on the server, however, most of the time it happens is when there is someone else in the server (this could be a complete coincidence too)

Playing in the server without OptiFine will not crash me at all, and playing OptiFine in singleplayer doesn't crash me, even when trying shaders.
However, it did crash me once when trying to enter the singleplayer world, but couldn't replicate it after that.

## Steps to Reproduce
1. Select and play with 1.16.5-OptiFine_HD_U_G7
2. Play multiplayer on Aternos Server.
3. Either change, add or remove resource packs and there probably will be a crash.
4. Alternatively, the game may crash for seemingly no reason.

## OptiFine Version
1.16.5-OptiFine_HD_U_G7

## Installation Method
Used the included standalone installer.

## Log Files/Crash Reports
[Latest Crash.txt](https://github.com/sp614x/optifine/files/6044096/Latest.Crash.txt)

[2021-02-25-3.log.gz](https://github.com/sp614x/optifine/files/6044099/2021-02-25-3.log.gz)

## F3 Debug Screenshot
![image](https://user-images.githubusercontent.com/67271557/109178760-cfa84d00-774e-11eb-80ee-5f87e4bc27a8.png)

(This one was taken after changing a resource pack without crashing)
![image](https://user-images.githubusercontent.com/67271557/109179001-10a06180-774f-11eb-9d49-4a4a49f88608.png)

## Prior Testing
If you've installed standalone:
- Does this happen in Vanilla without OptiFine? No.

## Additional Information
I play with a lot of OptiFine dependent resourcepacks, the world of the server also has a bunch of datapacks. In case this is relevant for something. Hope that the resources I provided made clear what the problem could be.

**Comment by KaiAF** [comment]
Does this happen on singleplayer? And when I went and tested it, seems like it didn't crash. 
Maybe its a shader, or maybe there is a weird block on the server that is causing the crash.

**Comment by HauntedSniper** [comment]
Ok, I just checked, and apparently, the problem seems to be that the server trying to load the resource pack would make it crash. I deleted the server resource pack link from the aternos page and now that it didn't try to load it, everything went right. Even adding and moving resourcepacks from the game would no longer crash Minecraft. 

(I hope I explained myself as I said the word "resourcepack" a lot)

Hopefully, now I would be able to play with optifine properly. It didn't seem to crash or get all weird in the few minutes I played on the server. It's still something weird, as 1.16.4 optfine didn't have this problem for me, but it shouldn't be a problem anymore, for the moment.
